x-pack/plugins/security_solution/public/overview/components/detection_response/rule_alerts_table/rule_alerts_items.ts

```diff
@@ -0,0 +1,165 @@
+  size: 0,
+  query: {
+    bool: {
+      filter: [
+        { term: { 'kibana.alert.workflow_status': 'open' } },
+        { range: { '@timestamp': { gte: from, lte: to } } },
+      ],
+    },
+  },
+  aggs: {
+    alertsByRule: {
+      terms: {
+        // top 4 rules sorted by severity counters
+        field: 'kibana.alert.rule.uuid',
+        size: 4,
+        order: [{ critical: 'desc' }, { high: 'desc' }, { medium: 'desc' }, { low: 'desc' }],
+      },
+      aggs: {
+        // severity aggregations for sorting
+        critical: { filter: { term: { 'kibana.alert.severity': 'critical' } } },
+        high: { filter: { term: { 'kibana.alert.severity': 'high' } } },
+        medium: { filter: { term: { 'kibana.alert.severity': 'medium' } } },
+        low: { filter: { term: { 'kibana.alert.severity': 'low' } } },
+        // get the newest alert to extract timestamp and rule name
+        lastRuleAlert: {
+          top_hits: {
+            size: 1,
+            sort: {
+              '@timestamp': 'desc',
+            },
+          },
+        },
+      },
+    },
+  },
+});
+
+const getRuleAlertsItemsFromAggs = (
+  aggregations?: SeverityRuleAlertsAggsResponse
+): RuleAlertsItem[] => {
+  const buckets = aggregations?.alertsByRule.buckets ?? [];
+  return buckets.map<RuleAlertsItem>((bucket) => {
+    const lastAlert = bucket.lastRuleAlert.hits.hits[0]._source;
+    return {
+      id: bucket.key,
+      alert_count: bucket.lastRuleAlert.hits.total.value,
+      name: lastAlert['kibana.alert.rule.name'],
+      last_alert_at: lastAlert['@timestamp'],
+      severity: lastAlert['kibana.alert.severity'],
+    };
+  });
+};
+
+export const useRuleAlertsItems = ({
+  queryId,
+  signalIndexName,
+  skip = false,
+}: {
+  queryId: string;
+  signalIndexName: string | null;
+  skip?: boolean;
+}) => {
+  const { to, from, deleteQuery, setQuery } = useGlobalTime();
+
+  const {
+    loading: isLoading,
+    data,
+    setQuery: setAlertsQuery,
+    response,
+    request,
+    refetch: refetchQuery,
+  } = useQueryAlerts<{}, SeverityRuleAlertsAggsResponse>({
+    query: getSeverityRuleAlertsQuery({
+      from,
+      to,
+    }),
+    indexName: signalIndexName,
+    skip,
+  });
+
+  const items = useMemo(() => {
+    if (data == null) {
+      return [];
+    }
+    return getRuleAlertsItemsFromAggs(data.aggregations);
+  }, [data]);
+
+  useEffect(() => {
+    setAlertsQuery(
+      getSeverityRuleAlertsQuery({
+        from,
+        to,
+      })
+    );
+  }, [setAlertsQuery, from, to]);
+
+  const refetch = useCallback(() => {
+    if (!skip && refetchQuery) {
+      refetchQuery();
+    }
+  }, [skip, refetchQuery]);
+
+  useQueryInspector({
+    deleteQuery,
+    inspect: {
+      dsl: [request],
+      response: [response],
+    },
+    refetch,
+    setQuery,
+    queryId,
+    loading: isLoading,
+  });
+
+  return { items, isLoading, queryId };
```

Comment:
I dont think you need to reexport `queryId` anymore since you define it in at the table level